pkg/builder/labels_annotations.go

```diff
@@ -88,6 +88,12 @@ func makeLabelsForObject(vdb *vapi.VerticaDB, sc *vapi.Subcluster, forPod bool)
 
 // MakeLabelsForPodObject constructs the labels that are common for all pods
 func MakeLabelsForPodObject(vdb *vapi.VerticaDB, sc *vapi.Subcluster) map[string]string {
+	return makeLabelsForObject(vdb, sc, true)
```

Comment:
I think the test was failing because when the replicaB is sandboxed, the sts podTemplateSpec is updated(sandbox name label added) which causes a pod replacement with a rollingUpdateStrategy. That's why `*-pri1-sb-2` is restarted(and will never be up again as restart reconciler is never called) and we get `upNodeCount mismatch expect 3 actual 2`. I don't think we really want to rolling update those pods, don't we?
So if we want to add sandbox label to the pods, maybe we should do so after the sandbox reconciler. What do you think?

Reply:
Yeah, makes sense. Since the pods have to be restarted by the sandbox controller, it seems like it should apply the label then.